### DISCUSSION TITLE
如何将extra中的内容提取为自定义的字段？

### DISCUSSION THREAD

**Discussion by CoLin00** [aidenlx/obsidian-zotero#215] [Help]
您好，感谢大佬的插件，非常好用！

请问下大佬，我如何可以自定义提取extra的内容。

extra中我输入了"original-date: 1997"，我在模版中设定了：

`original-date: <%=  /original-date:\s*(\d+)/.exec(it.extra)[1]  %>`

当我的文献中extra中没有输入original-date相关信息的时候，这个命令会出错，模版无法执行。在有这个数据的时候，输出是正常的。

另外，若是extra中我添加了多行的类似数据应该如何执行呢？例如，当extra中有：“Original Author: James C. Scott
Original Title: Seeing Like a State: How Certain Schemes to Improve the Human Condition Have Failed”这样的内容的时候呢？

谢谢！

**Comment by aidenlx** [comment]
```ejs
<% const ogDate = it.extra?.match(/original-date:\s*(\d+)/)?.[1]; if (ogDate) { %>
original-date: <%= ogDate %>
<% } %>
```

**Reply by CoLin00** [reply]
感谢，但是不知为何这行代码也无法输出。

**Reply by aidenlx** [reply]
报错发出来下？

**Reply by CoLin00** [reply]
[ERROR] obsidian-zotero:main - Failed to renderYaml it.extra?.match is not a function {

**Comment by aidenlx** [comment]
v2 会直接解析 Extra 字段，不用再写正则了。`zt.extra.fields` 是按键名索引的对象：

```liquid
original-date: {{ zt.extra.fields["original-date"] }}
```

键不存在时读出来是空值，不会再让整个模板报错。多行的 `Original Author` / `Original Title` 同理，各自按键名取即可，互不影响。

v2 beta 现已发布：https://zotlit.aidenlx.site/docs/install-zotlit
迁移指南：https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
